VTKU-110 : ammatillisten arvosanojen leikkuripvm

timorantalaiho opened the pull request:
Ei mergattavaksi vielä.

* huomioidaan vain sellaiset ammetilliset perustutkinnot, joiden vahvistumispvm on parametrina annetun valmistumisen takarajan päivä tai aikaisempi
* kaadetaan laskenta, jos siihen päätyy opiskeluoikeuksia, joiden aikaleima on myöhemmin kuin parametrina annetun "data Koskessa viimeistään" -leikkuripäivämääränä.

Oikean dataversion kaivelu on vielä hieman työn alla Valintalaskentakoostepalvelun puolella.

timorantalaiho committed:
Poista tarpeettomaksi jäänyt Jetty-runneri

timorantalaiho committed:
VTKU-110 : Päivitä versionumerot

hsalokor committed:
WIP: kaavatesti

timorantalaiho committed:
Poista tarpeeton koodi

timorantalaiho committed:
VTKU-110 : Lue valmistumisraja parametrista

Toistaiseksi se on vielä kovakoodattu.

timorantalaiho committed:
VTKU-110 : Siistimpi logitus

hsalokor committed:
VTKU-103: Konvertoi reformin mukainen laskentakaavarakenne testiin

timorantalaiho committed:
VTKU-110 : Uploadaa Travisiin branchista

timorantalaiho committed:
VTKU-110 : Mahdollista tutkinnon rajapvm:n syöttäminen

hsalokor committed:
VTKU-103: Poista kommentti käsilaskennan jälkeen

timorantalaiho committed:
VTKU-110 : Käytä parametreja valintaperusteen sijaan

timorantalaiho committed:
VTKU-110 : Vahdi leikkuripäivää laskennassa

timorantalaiho committed:
VTKU-110 : Huomioi valmistumisen raja

hsalokor committed:
VTKU-103: Korjaa testiin review-parannuksia

timorantalaiho committed:
VTKU-110 : Raportoi vahvistuspäivämäärä

timorantalaiho committed:
Irrota yleiskäyttöinen metodi

Ei toiminnallisia muutoksia

timorantalaiho committed:
Poista toistoa

Ei toiminnallisia muutoksia

timorantalaiho committed:
Käytä commons-lang-3:n StringUtils:ia

timorantalaiho committed:
Varaudu myös null-stringeihin

timorantalaiho committed:
Korjaa muotoilu

Ei toiminnallisia muutoksia.

timorantalaiho committed:
VTKU-110 : Estä leikkuripvm:n ylitys

timorantalaiho committed:
Konvertoi Scalaksi vain kerran

Ei toiminnallisia muutoksia.

timorantalaiho committed:
VTKU-110 : Logita laskennassa käytettävät pvm:t

timorantalaiho committed:
VTKU-110 : Lisää testit pvm-logiikalle

rahcola commented on valintaperusteet-laskenta/src/main/scala/fi/vm/sade/kaava/Laskentadomainkonvertteri.scala:
```diff
@@ -660,24 +671,51 @@ object Laskentadomainkonvertteri {
+            val viesti: String = s"Ei pystytty tulkitsemaan suomalaista päivämäärää (esim '${suomalainenPvmMuoto.format(LocalDate.now())}' " +
+              s"syötteestä ${parametri.getArvo} funktion ${parametri.getFunktiokutsu.getFunktionimi} parametriksi '${parametri.getAvain}'. Käytetään oletusarvoa ${suomalainenPvmMuoto.format(oletusarvo)}"
+            LOG.error(viesti, e)
+            oletusarvo
```
Tän oletusarvon voisi ehkä myös merkitä TODO kommentilla poistettavaksi jahka parametrit on saatu syötettyä kaavoihin.

rahcola commented on valintaperusteet-laskenta/src/main/scala/fi/vm/sade/service/valintaperusteet/laskenta/LukuarvoLaskin.scala:
```diff
@@ -299,7 +299,7 @@ protected[laskenta] class LukuarvoLaskin(protected val laskin: Laskin)
         }
 
-      case f@IteroiAmmatillisetTutkinnot(_, _, _, _, _, _, _) => iteroiAmmatillisetTutkinnot(iteraatioParametrit, f)
+      case f@IteroiAmmatillisetTutkinnot(_, _, _, _, _, _, _, _, _) => iteroiAmmatillisetTutkinnot(iteraatioParametrit, f)
```
Olettaisin, että näissä voisi sanoa vähän lyhemmin `case f: IteroiAmmatillisetTutkinnot => ...`

timorantalaiho committed:
VTKU-110 : Lisää TODOt

timorantalaiho committed:
VTKU-110 : Yksinkertaista

Ei toiminnallisia muutoksia.

timorantalaiho committed:
VTKU-110 : Välitä oikea parametri eteenpäin

Ilmeisesti muun kuin tutkintojen laskentaa on käyttänyt
oletusparametria.

Tämän ei pitäisi kuitenkaan olla vaikuttanut
toiminnallisuuteen, koska on huomioitu vain leikkuripäivä-
määrään mennessä valmistuvat tutkinnot, ja tehty
iteraatioparametrit niiden perusteella.

timorantalaiho committed:
VTKU-110 : Logita kaikki tutkinnot

Nyt vasta iterointifunktion laskenta tiputtaa pois ne
tutkinnot, joista ei olla valmistumassa ajoissa.

timorantalaiho committed:
Päivitä riipopuvuuksia

timorantalaiho committed:
VTKU-110 : Salli vaihtuvat desimaalit

Ilmeisesti joidenkin Koski-opiskeluoikeuksien
aikaleima tulee vain kolmen desimaalin tarkuudella
kuuden sijasta.

timorantalaiho committed:
VTKU-110 : Paremmat nimet

timorantalaiho committed:
VTKU-110 : Salli tyhjät osasuorituksen perustiedot

Sain valintaryhmälaskennan kaatumaan tähän testatessa
oikealla datalla.

timorantalaiho committed:
VTKU-110 : Siirrä linssit siistimpään paikkaan

Ei toiminnallisia muutoksia.

timorantalaiho committed:
VTKU-110 : Poista käyttämätöntä koodia

Ei toiminnallisia muutoksia.

timorantalaiho committed:
VTKU-110 : Varmista ettei osa-alueen koodi haittaa

timorantalaiho committed:
VTKU-110 : Lisää koodi YTOn valintaan

Ilmeisesti suoraan suorituksen alla on normaalisti YTO-
koodilla varustettuja osasuorituksia vain silloin, kun ne
todella ovat tutkinnon osia, mutta ehkä tämä myös
selkeyttää vähän.

timorantalaiho committed:
VTKU-110 : Versiot masteria varten

timorantalaiho committed:
VTKU-110 : Branch-deploy pois
